packages/driver/src/cypress/error_messages.coffee

```diff
@@ -915,6 +924,7 @@ module.exports = {
     invalid_month: "Typing into a month input with #{cmd('type')} requires a valid month with the format 'yyyy-MM'. You passed: {{chars}}"
     invalid_week: "Typing into a week input with #{cmd('type')} requires a valid week with the format 'yyyy-Www', where W is the literal character 'W' and ww is the week number (00-53). You passed: {{chars}}"
     invalid_time: "Typing into a time input with #{cmd('type')} requires a valid time with the format 'HH:mm', 'HH:mm:ss' or 'HH:mm:ss.SSS', where HH is 00-23, mm is 00-59, ss is 00-59, and SSS is 000-999. You passed: {{chars}}"
+    invalid_dateTime: "Typing into a datetime input with #{cmd('type')} requires a valid datetime with the format 'yyyy-MM-ddThh:mm', for example '2017-06-01T08:30'. You passed: {{chars}}"
     multiple_elements: "#{cmd('type')} can only be called on a single element. Your subject contained {{num}} elements."
     not_on_typeable_element: """
```

Comment:
refactor these error messages like we said - listing out what we consider 'isFocusable' elements - throw a more specific error message if the element is `[disabled]` and then list the other elements we check (with disabled excluded) for the error message if not already caught.

Reply:
This may seem very naive since I've spent barely any time with the implementation. I'm concerned about the kind of tests the current API allows with regard to accessibility:

Why do you even check focusability? The user agent already takes care of this and any custom logic you have will require maintenance. Simply forwarding `type` to something like `document.activeElement.dispatchEvent` seems like a simpler implementation that isn't subject to any confusion as to which elements are able to receive keyboard events.

The testing API shouldn't allow specifying the element that receives the keyboard event in the first place (or only when enabling a nasty sounding flag like `unsafe` or `force`). Otherwise you can construct test cases that are unreachable by a  real user.

Reply:
>The user agent already takes care of this

I'm not sure what you mean by this. As browser automation software, Cypress has to programmatically automate everything, there is not a real user clicking, etc. Knowing what allows a focus event is a necessary part of any browser automation software. You're maybe thinking that Native browser events support would be better? https://github.com/cypress-io/cypress/issues/311 We agree and this is one of the requirements for moving forward with that.

Reply:
Hm maybe I'm misunderstanding something about the architecture but you do use the DOM provided by Chrome or electron no?

These user agents already prevent focus on an element that can't receive focus e.g.

```html
<main>try focusing me</main>
```

```js
console.assert(document.activeElement === document.body);

const main = document.querySelector('main');
main.focus();

console.assert(document.activeElement === document.body);
```
does already work. You do not have to implement any custom logic to only allow focus on e.g. elements with tabindex or `<button />` etc. So if you do have access to the DOM of a user agent such as chrome or electron there's no need for a custom focus filter.

Reply:
> Why do you even check focusability? The user agent already takes care of this and any custom logic you have will require maintenance.

@eps1lon this is a very common misconception that the user agent will perform default actions (inserting text, moving focus, moving cursor) by using `dispatchEvent`. This isn't true, and we must implement the default actions ourselves. The only case where the browser will react to a synthetic event sent through `dispatchEvent` is a checkbox can be toggled with the `click` event.

Reply:
@eps1lon also, when the browser is out of focus, `document.activeElement` will be updated, but focus events are queued until the window focus is regained. Because of this we need to "polyfill" focus events in some cases

Reply:
> this is a very common misconception that the user agent will perform default actions (inserting text, moving cursor) by using dispatchEvent.

I understand that. I'm specifically talking about `HTMLElement.focus()` not `dispatchEvent(new FocusEvent())`. See https://github.com/cypress-io/cypress/pull/4870#discussion_r328651245